seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/Firefox.scala

```diff
@@ -33,10 +36,21 @@ object Firefox extends SeleniumBrowser {
         |    oldErr.apply(console, arguments);
         |  };
         |  this.scalajsPopHijackedConsoleLog = function () {
-        |    var log = console_log_hijack;
-        |    if (log.length != 0)
-        |      console_log_hijack = [];
-        |    return log;
+        |    if (console_log_hijack.length == 0) {
+        |      return console_log_hijack;
+        |    } else {
+        |      var log = [];
+        |      while (currentLogIndex < console_log_hijack.length &&
+        |          log.length < 1024) {
+        |        log.push(console_log_hijack[currentLogIndex]);
+        |        currentLogIndex++;
+        |      }
+        |      if (console_log_hijack.length == currentLogIndex) {
+        |        console_log_hijack = [];
+        |        currentLogIndex = 0;
+        |      }
```

Comment:
That was my first approach, but modifying `console_log_hijack` has linear time and hence ends up timing out in the same cases that the previous solution did. Additionally, each time the console logs are fetched `scalajsPopHijackedConsoleLog` is called in a loop until the result is empty, at which point the `console_log_hijack` is reinitialized.